src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.Small.cs

```diff
@@ -30,33 +30,49 @@ protected override bool ReadAndCacheConstructorArgument(ref ReadStack state, ref
             switch (jsonParameterInfo.Position)
             {
                 case 0:
-                    success = ((JsonParameterInfo<TArg0>)jsonParameterInfo).ReadJsonTyped(ref state, ref reader, out TArg0 arg0);
-                    if (success)
                     {
-                        arguments.Arg0 = arg0!;
+                        var info = (JsonParameterInfo<TArg0>)jsonParameterInfo;
+                        var converter = (JsonConverter<TArg0>)jsonParameterInfo.ConverterBase;
+                        success = converter.TryRead(ref reader, info.RuntimePropertyType, info.Options, ref state, out TArg0 arg0);
```

Comment:
Yes perhaps it would save a bit for the single reference-type-compiled version (but not for every value-type compilation). It will also make this more readable.